engine/common/rpc/convert/execution_data.go

```diff
@@ -199,10 +229,6 @@ func messageToTrustedCollection(
 		transactions[i] = &transaction
 	}
 
-	if len(transactions) == 0 {
-		return nil, nil
```

Comment:
this is incorrect. EN, VNs and badger will return a collection with an empty set of transactions. Returning `nil` here would result in different results when converting to and from grpc entities.